### PR TITLE
language-based goal specification parsed with LLMs

### PR DESCRIPTION
**Pull request by tomsilver**
To be merged after #1364 

For example, given a directory `blocks_tasks` with a JSON file `task1.json` that contains:
```json
{
    "problem_name": "blocks_language_test_problem",
    "blocks": {
        "red": {
            "position": [1.36409716, 1.0389289, 0.2225],
            "color": [1, 0, 0]
        },
        "green": {
            "position": [1.36409716, 1.0389289, 0.2675],
            "color": [0, 1, 0]
        },
        "blue": {
            "position": [1.35479861, 0.91064759, 0.2225],
            "color": [0, 0, 1]
        }
    },
    "block_size": 0.045,
    "language_goal": "Make a tower with red on green and green on blue."
}
```

Then running 

```python predicators/main.py --env blocks --approach oracle --seed 0 --test_task_json_dir blocks_tasks --num_test_tasks 1 --make_test_videos```

works and produces this video:
https://user-images.githubusercontent.com/2816502/216443485-aef93973-0e1c-4ca0-8197-c4c26d44eadf.mp4

Also works with PyBullet:

```python predicators/main.py --env pybullet_blocks --approach oracle --seed 0 --test_task_json_dir blocks_tasks --num_test_tasks 1 --make_test_videos --use_gui --video_fps 15```

https://user-images.githubusercontent.com/2816502/216443650-c95f5c60-04ed-48c8-9adc-4d4201b1e8c3.mp4

### CODE REVIEW

**Comment by tomsilver** [comment]
@williamshen-nz you don't actually need to review, just tagging in case you're interested

File: predicators/envs/blocks.py [hunk 1]
```diff
@@ -588,12 +588,24 @@ def _load_task_from_json(self, json_file: Path) -> Task:
...
+            goal = self._parse_language_goal_from_json(
+                task_spec["language_goal"], id_to_obj)
         else:
             raise ValueError("JSON task spec must include 'goal'.")
         task = Task(init_state, goal)
         assert not task.goal_holds(init_state)
         return task
 
+    def _get_language_goal_prompt_prefix(self) -> str:
+        return """
```
Comment (wmcclinton) [inline predicators/envs/blocks.py:601]: Weird format. We can't tab?

**Review by wmcclinton — APPROVED** [review]
Looks Great!

**Comment by tomsilver** [comment]
for extra fun, I recorded an audio file

[blocks_goals.m4a.zip](https://github.com/Learning-and-Intelligent-Systems/predicators/files/10573393/blocks_goals.m4a.zip)

and then uploaded it to https://replicate.com/openai/whisper

which correctly transcribed to "Make a tower with red on green and green on blue."

so that will be a fun part of the eventual demo :)

**Review by NishanthJKumar — APPROVED** [review]
LGTM! Super exciting to have this implemented :tada:

**Pull request merged**